### ISSUE TITLE
"Align parameters and components" when using multiple indices in nested table expression

### ISSUE THREAD

**Issue by ConjuringCoffee** [SAP/abap-cleaner#75]
Hi Jörg-Michael, I found an example in which I don't like how alignment looks when working with multiple indices in a table expression with a nested table. Any chance you can improve this? 😊

Take the following full example:

```abap
CLASS lcl_example DEFINITION CREATE PRIVATE.

  PUBLIC SECTION.
    METHODS example.
  PROTECTED SECTION.
  PRIVATE SECTION.
    METHODS this_is_a_very_very_long_name
      IMPORTING
        i_import        TYPE bapiret2
      RETURNING
        VALUE(r_result) TYPE string.
ENDCLASS.


CLASS lcl_example IMPLEMENTATION.
  METHOD example.
    DATA table_normal TYPE bapiret2_t.
    DATA table_nested TYPE TABLE OF bapiret2_t.

    DATA(this_is_a_very_very_long__name) = NEW lcl_example( )->this_is_a_very_very_long_name( table_normal[ 1 ] ).

    DATA(this_is_a_very_very_long_name2) = NEW lcl_example( )->this_is_a_very_very_long_name( table_nested[ 1 ][ 1 ] ).
  ENDMETHOD.

  METHOD this_is_a_very_very_long_name.
  ENDMETHOD.
ENDCLASS.
```

I'm not really happy with the indentation of the table expression. This is the result after ABAP Cleaner was executed:

```abap
    DATA(this_is_a_very_very_long__name) = NEW lcl_example( )->this_is_a_very_very_long_name(
                                                   table_normal[ 1 ] ).

    DATA(this_is_a_very_very_long_name2) = NEW lcl_example( )->this_is_a_very_very_long_name( table_nested[
                                                                                                  1 ][ 1 ] ).
```

The first example is alright, but I don't like the second example. I'd have expected something like this:

```abap
    DATA(this_is_a_very_very_long_name2) = NEW lcl_example( )->this_is_a_very_very_long_name(
                                                   table_nested[ 1 ][ 1 ] ).
```

If I tried to manually indent it like that though, then this would be the result:

```abap
    DATA(this_is_a_very_very_long_name2) = NEW lcl_example( )->this_is_a_very_very_long_name(
                                                                                              table_nested[
                                                                                                  1 ][ 1 ] ).
```

Here are the settings I use:
![image](https://github.com/SAP/abap-cleaner/assets/72548231/9fd00ef4-49d3-44c4-b90e-ec3641b8559c)

**Comment by jmgrassau** [comment]
Hi ConjuringCoffee, 

yes, that second example definitely doesn't look good (although in a way it's not wrong). On the other hand, if you had a variable name here such as ```[ lv_index ]``, it could have any length between 1 and 30 chars, and at some point you'd want a line break. So, let's see what can be done about it – maybe it should simply be prevented to introduce line breaks when the content is very short. 

What I'd consider a bug is the last example, because if ABAP cleaner moves ```table_nested[``` behind the opening ```(```, there is no reason to keep the line break. I'll have to check why that happens!

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi ConjuringCoffee, 

thanks again for opening this issue – this was indeed a bug, because table expressions that are chained with ```][``` were not processed properly. With your example and settings, I now get:

```ABAP
    DATA(this_is_a_very_very_long__name) = NEW lcl_example( )->this_is_a_very_very_long_name(
                                                   table_normal[ 1 ] ).

    DATA(this_is_a_very_very_long_name2) = NEW lcl_example( )->this_is_a_very_very_long_name(
                                                   table_nested[ 1 ][ 1 ] ).
```

Condensing also works now (without keeping unnecessary line breaks), so if you enter …

```ABAP
    any_method(
    lt_any_table[ 
    1 ][ 
    2 ][ 
    3 ] ).

    any_method(
    par1 = lt_any_table[ 
    1 ]
    param2 = lt_any_table[ 
    1 ][ 
    2 ]
    parameter3 = lt_any_table[ 
    1 ][ 
    2 ][ 
    3 ] ).
```

you will get: 

```ABAP
    any_method( lt_any_table[ 1 ][ 2 ][ 3 ] ).

    any_method( par1       = lt_any_table[ 1 ]
                param2     = lt_any_table[ 1 ][ 2 ]
                parameter3 = lt_any_table[ 1 ][ 2 ][ 3 ] ).
```

And an example where a line break cannot be avoided …

```ABAP
    DATA(lv_some_value_with_a_long_name) = any_method(
    lt_any_table[
    lv_index ][
    lv_second_index ][
    lv_third_index_with_long_name ] ).
```

… will look like this:

```ABAP
    DATA(lv_some_value_with_a_long_name) = any_method( lt_any_table[ lv_index ][ lv_second_index ][
                                                           lv_third_index_with_long_name ] ).
```

Kind regards,
Jörg-Michael

**Comment by ConjuringCoffee** [comment]
Great, thank you!

**Comment by jmgrassau** [comment]
Hi ConjuringCoffee, 

this issue should now be fixed with version 1.5.1!

Kind regards,
Jörg-Michael
